akka-docs/src/main/paradox/java/stream/stream-customize.md

```diff
@@ -1,414 +1 @@
-lead to calling `pull`.
-
-One of the important use-case for this is to build buffer-like entities, that allow independent progress
-of upstream and downstream stages when the buffer is not full or empty, and slowing down the appropriate side if the
-buffer becomes empty or full.
-
-The next diagram illustrates the event sequence for a buffer with capacity of two elements in a setting where
-the downstream demand is slow to start and the buffer will fill up with upstream elements before any demand
-is seen from downstream.
-
-![graph_stage_detached_tracks_1.png](../../images/graph_stage_detached_tracks_1.png)
-
-Another scenario would be where the demand from downstream starts coming in before any element is pushed
-into the buffer stage.
-
-![graph_stage_detached_tracks_2.png](../../images/graph_stage_detached_tracks_2.png)
-
-The first difference we can notice is that our `Buffer` stage is automatically pulling its upstream on
-initialization. The buffer has demand for up to two elements without any downstream demand.
-
-The following code example demonstrates a buffer class corresponding to the message sequence chart above.
-
-@@snip [GraphStageDocTest.java]($code$/java/jdocs/stream/GraphStageDocTest.java) { #detached }
-
-## Thread safety of custom processing stages
-
-All of the above custom stages (linear or graph) provide a few simple guarantees that implementors can rely on.
-: 
- * The callbacks exposed by all of these classes are never called concurrently.
- * The state encapsulated by these classes can be safely modified from the provided callbacks, without any further
-synchronization.
-
-
-In essence, the above guarantees are similar to what `Actor` s provide, if one thinks of the state of a custom
-stage as state of an actor, and the callbacks as the `receive` block of the actor.
-
-@@@ warning
-
-It is **not safe** to access the state of any custom stage outside of the callbacks that it provides, just like it
-is unsafe to access the state of an actor from the outside. This means that Future callbacks should **not close over**
-internal state of custom stages because such access can be concurrent with the provided callbacks, leading to undefined
-behavior.
-
-@@@
-
-## Resources and the stage lifecycle
-
-If a stage manages a resource with a lifecycle, for example objects that need to be shutdown when they are not
-used anymore it is important to make sure this will happen in all circumstances when the stage shuts down.
-
-Cleaning up resources should be done in `GraphStageLogic.postStop` and not in the `InHandler` and `OutHandler`
-callbacks. The reason for this is that when the stage itself completes or is failed there is no signal from the upstreams
-or the downstreams. Even for stages that do not complete or fail in this manner, this can happen when the
-`Materializer` is shutdown or the `ActorSystem` is terminated while a stream is still running, what is called an
-"abrupt termination".
+../../scala/stream/stream-customize.md
```

Comment:
Seems something went wrong here - the file was replaced by a file containing the contents "../../scala/stream/stream-customize.md", but it should have been replaced with a symlink. The difference is a bit hard to see in the github UI but should be clear when you build the docs locally. Could you make this a symlink?